server/log_operation_manager.go

```diff
@@ -158,14 +171,16 @@ func (er *electionRunner) Run(ctx context.Context) {
 				break
 			}
 			if er.shouldResign(masterSince) {
-				glog.Infof("%d: deliberately resigning mastership", er.logID)
+				glog.Infof("%d: queue up resignation of mastership", er.logID)
```

Comment:
The commonization would be nice, but it would mean that the mastership tracker (and the metric) would stay `true` until the outer code processed the pending resignation; it feels safer to update that immediately.